Cabal/src/Distribution/Simple/GHC.hs

```diff
@@ -710,7 +710,10 @@ buildOrReplLib mReplFlags verbosity numJobs pkg_descr lbi lib clbi = do
     let (cSrcs', others) = partition (\filepath -> ".c"`isSuffixOf` filepath) (cSources libBi)
     unless (null others) $ do
       let files = intercalate ", " others
-      warn verbosity $ "The following files listed in c-sources will not be used: " <> files
+      let libraryName = case libName lib of
```

Comment:
We have [`showLibraryName`](https://hackage.haskell.org/package/Cabal-syntax-3.10.1.0/docs/Distribution-Types-LibraryName.html#v:showLibraryName)

Reply:
The output is unsatisfying. Especially because it is:

> "library '" ++ prettyShow name ++ "'"

So the output would be:

> "The following files listed in library 'lib2''s c-sources  will not be used:

The double single quotes are not a mistake.

And if it's the main library, it only outputs "library", which breaks the flow of the sentence.

Reply:
Is
```
libraryNameStanza :: LibraryName -> String
libraryNameStanza LMainLibName       = "library"
libraryNameStanza (LSubLibName name) = "library " ++ prettyShow name
```
any better? Otherwise, it's ok.

Reply:
Or rephrase as "The following files listed in the c-sources of library 'lib2' will not be used"

Reply:
@fgaz I don't understand what this suggestion is addressing, sorry

Reply:
If you phrase it like that, you can use `showLibraryName`

Reply:
But we lose the "main" bit, which is fairly important.

Reply:
Good point. I wonder if `showLibraryName` produces confusing messages elsewhere too. Maybe we should add "main" there too...

Reply:
Yes indeed, we will have to formalise a bit more the presentation layer for cabal-install.